server.coffee

```diff
@@ -299,6 +299,19 @@ render_template_with_context = (ctx) ->
       resolve ctx
   )["catch"]((err) -> ctx.error = err; Promise.reject ctx)
 
+inject_template_name = (ctx) ->
+  new Promise( (resolve, reject) ->
+      ###
+      if ctx.rendered_template.indexOf('EXEC sp_executesql') != -1
+        log.debug "******** Parameterized query detected do something smart here *************"
+      else
+      ###
+      arr = ctx.rendered_template.split('\n')
+      arr.unshift('-- ' + ctx.template_name)
+      ctx.rendered_template = arr.join('\n')
```

Comment:
if we just want to do that couldn't we also just do it in the render_template_with_context method and not even bother with another function...until we need to?